test/validation/api/ipsec/ipsec.h

```diff
@@ -82,12 +86,6 @@ typedef struct {
 		odp_proto_l4_type_t l4_type;
 		uint32_t seq_num;
 	} out[1];
```

Comment:
While we are at it, do you think this out[1] should be continued? If the test_part is supposed to indicate one packet and the details associated with it, num_pkt and this [1] looks redundant to me.

With reassembly spec change, I did submit a change in increasing the array size, but now that I redid some of the code, I'm starting to think that these logic can stay in the caller and the test part can be independent. 

I would assume the following with reassembly,

Field, pkt_in can be NULL. In that case no packet to be send. Intention of test_part would be to hold the resultant output packet.
Field ipsec_test_part.out[1].pkt_res can be NULL, then no packet to be received. Intention of test_part to just transmit out the packet.

Of course, I won't be making these assumptions in the code. I intend to move all logic to callers rather than make decisions based on ipsec_test_part. But the above would be the overall angle that I would take. Just wanted to confirm that with reassembly one input packet may not have an output (yet). And one output packet (one among the partial reassembly case) may not have a corresponding input packet.

If I'm not clear, we can defer this discussion when I revisit reassembly.

Reply:
> While we are at it, do you think this out[1] should be continued? If the test_part is supposed to indicate one packet and the details associated with it, num_pkt and this [1] looks redundant to me.

ipsec_test_part is about one ipsec *operation*. One IPsec operation for one packet can have output of multiple packets (in case of fragmentation offload) and therefore ipsec_test_part can hold one input packet and multiple output packets. The multiple-output case is not currently used but I did not want to strip it away.

Reply:
@JannePeltonen Okay. Then I will retain increasing the size of this array as required for reassembly case.